app/src/main/java/com/example/geoquiz/QuizActivity.kt

```diff
@@ -0,0 +1,196 @@
+package com.example.geoquiz
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+
+import android.widget.Button
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProviders
+
+
+private const val TAG = "MainActivity"
+private const val QUESTION_INDEX_KEY = "index"
+private const val QUESTIONS_ANSWERED_KEY = "answer"
+private const val REQUEST_CODE_CHEAT = 0
+
+class QuizActivity : AppCompatActivity() {
+
+
+    private lateinit var trueButton: Button
+    private lateinit var falseButton: Button
+    private lateinit var nextButton: ImageButton
+    private lateinit var prevButton: ImageButton
+    private lateinit var questionTextView: TextView
+    private lateinit var cheatButton: Button
+    private lateinit var scoreButton: Button
+
+
+    private val quizViewModel: QuizViewModel by lazy {
+        ViewModelProviders.of(this).get(QuizViewModel::class.java)
+    }
+
+
+    private var questionWasAnswered: BooleanArray = BooleanArray(4) //button true or false was clicked
+    private lateinit var correctAnswers: ArrayList<Int>
+    private var score: Int = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Log.d(TAG, "onCreate(Bundle?) called")
+        setContentView(R.layout.activity_quiz)
+
+
+
+        trueButton = findViewById(R.id.true_button)
+        falseButton = findViewById(R.id.false_button)
+        questionTextView = findViewById(R.id.question_text_view)
+        nextButton = findViewById(R.id.next_button)
+        prevButton = findViewById(R.id.prev_button)
+        scoreButton = findViewById(R.id.score_button)
+        cheatButton = findViewById(R.id.cheat_button)
+
+        if (savedInstanceState != null) {
+            quizViewModel.currentIndex = savedInstanceState.getInt(QUESTION_INDEX_KEY)
+            questionWasAnswered = savedInstanceState.getBooleanArray(QUESTIONS_ANSWERED_KEY)!!
+        }
+
+
+
+        questionTextView.setOnClickListener {
```

Comment:
Please create a method called `setClickListeners` and move all of them there. It makes the `onCreate` code easier to follow

Reply:
Do this for all activities